File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        maximum = 0
        hashSet = set(nums)
        for i in nums:
            if i-1 not in hashSet:
                minimal_number = i
                curr_streak = 0
                while minimal_number in hashSet:
                    minimal_number += 1
                    curr_streak += 1
                maximum = max(maximum, curr_streak)
        return maximum

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set(nums.begin(),nums.end());

        int longest_sequence = 0; 
        for(int i=0; i<nums.size(); i++){
            if(set.find(nums[i]-1) != set.end()){
                continue;
            }
            else{
                int curr_streak = 0;
                int current_element = nums[i];
                while(set.find(current_element) != set.end()){
                    curr_streak += 1;
                    current_element++;
                }
                
                longest_sequence = max(longest_sequence, curr_streak);
            }
        }
        return longest_sequence; 
    }
};